### PR TITLE
Filter out initial action created event if action created from program

### PR DESCRIPTION
**Pull request by luketlancaster**
- [x] Add the Clubhouse Story ID: [ch21350]

### CODE REVIEW

File: test/integration/patients/sidebar/action-sidebar.js [hunk 1]
```diff
@@ -511,7 +512,7 @@ context('action sidebar', function() {
 
     cy
       .get('[data-activity-region]')
-      .should('contain', 'RoundingWell (System) added this Action')
+      .should('not.contain', 'RoundingWell (System) added this Action')
       .should('contain', 'Clinician McTester (Nurse) changed the Owner to Another Clinician')
       .should('contain', 'Clinician McTester (Nurse) changed the details of this Action')
       .should('contain', 'Clinician McTester (Nurse) changed the Due Date to ')
```
Comment (luketlancaster) [inline test/integration/patients/sidebar/action-sidebar.js]: you think we need a whole 'nother set of tests for a non-program action?

Comment (paulfalgout) [reply test/integration/patients/sidebar/action-sidebar.js]: was that the only test for `RoundingWell (System)` cause I don't want to lose that.  But yeah we probably should have another set of tests for the other option.. 1 for code coverage, but 2 `.should('not.contain', 'Whatever')` would also pass.. really we could probably add a test for an action with a program and have 2 events.. one create and the other the pathway one and then check for length of 1.

File: src/js/views/patients/sidebar/action/action-sidebar-activity-views.js [hunk 2]
```diff
@@ -96,6 +96,10 @@ const ActivityView = View.extend({
 
 const ActivitiesView = CollectionView.extend({
   childView: ActivityView,
+  viewFilter({ model }) {
+    if (model.get('type') === 'ActionCreated' && this.model.getProgram()) return false;
```
Comment (paulfalgout) [inline src/js/views/patients/sidebar/action/action-sidebar-activity-views.js]: for some reason I assumed we already had a `getProgram` when slacking earlier.  We probably won't need it for anything.. we can just use `get('_program')` here and add less code.

File: test/integration/patients/sidebar/action-sidebar.js (hunk 1 above)
Comment (luketlancaster) [reply test/integration/patients/sidebar/action-sidebar.js]: alrighty, I added another round for 'created by program action'. take a look and lemme know

File: test/integration/patients/sidebar/action-sidebar.js [hunk 3]
```diff
@@ -202,7 +202,7 @@ context('action sidebar', function() {
       .should('not.exist');
   });
 
-  specify('display action sidebar', function() {
+  specify.only('display action sidebar', function() {
```
Comment (LisaManresa) [inline test/integration/patients/sidebar/action-sidebar.js]: remove .only

File: test/integration/patients/sidebar/action-sidebar.js [hunk 4]
```diff
@@ -246,7 +246,7 @@ context('action sidebar', function() {
       })
       .routePatientActions(fx => {
         fx.data[0] = actionData;
-
+        console.log(fx.data);
```
Comment (LisaManresa) [inline test/integration/patients/sidebar/action-sidebar.js]: rogue console.log

File: test/integration/patients/sidebar/action-sidebar.js (hunk 3 above)
Comment (luketlancaster) [reply test/integration/patients/sidebar/action-sidebar.js]: It is one way to make the tests run faster 🙃 (fixed in latest push)

Comment (LisaManresa) [reply test/integration/patients/sidebar/action-sidebar.js]: "QA engineers hate him. Find out this dev's brilliant secret to speeding up integration tests!"

**Pull request merged**